Source/DafnyCore/Verifier/ProofObligationDescription.cs

```diff
@@ -297,7 +297,7 @@ public class AssertStatement : ProofObligationDescription {
       : $"error is impossible: {customErrMsg}";
 
   public override string FailureDescription =>
-    customErrMsg ?? "assertion might not hold";
+    customErrMsg ?? "Could not prove assertion";
```

Comment:
```suggestion
    customErrMsg ?? "could not prove assertion";
```

The convention in the rest of the messages is for them to start with a lower-case letter.